retrieval/targetmanager.go

```diff
@@ -106,7 +107,7 @@ func (tm *TargetManager) reload() {
 			ts = &targetSet{
 				ctx:    ctx,
 				cancel: cancel,
-				sp:     newScrapePool(ctx, scfg, tm.append, tm.logger),
+				sp:     newScrapePool(ctx, scfg, tm.append, log.With(tm.logger, "scrape_pool", scfg.JobName)),
```

Comment:
Yea, I considered both. Ultimately it's the problem of `job` not necessarily being the final job.